benchexec/tablegenerator/react-table/src/components/FilterBox/FilterCard.js

```diff
@@ -103,11 +111,39 @@ export default class FilterCard extends React.PureComponent {
     };
     const [vMin, vMax] = value.split(":");
     return {
-      min: vMin.trim() !== "" ? builder(vMin) : builder(propMin),
-      max: vMax.trim() !== "" ? builder(vMax) : builder(propMax),
+      min: vMin.trim() !== "" ? vMin : builder(propMin),
+      max: vMax.trim() !== "" ? vMax : builder(propMax),
     };
   }
 
+  handleNumberChange(min, max) {
+    const newState = {};
+    newState.sliderMin = Number(
+      this.state.numericMin !== null
+        ? this.state.numericMin
+        : this.state.sliderMin,
+    );
+    newState.sliderMax = Number(
+      this.state.numericMax !== null
+        ? this.state.numericMax
+        : this.state.sliderMax,
+    );
```

Comment:
Out of curiosity mostly: Our build process desugars this to
```
sliderMin=Number(null!==(n=this.state.numericMin)&&void 0!==n?n:this.state.sliderMin)
```
Does `void 0!==n` change the semantics compared to the original code, for example when `numericMin` is 0? (I have trouble interpreting it because of the `void`.)

Reply:
Well yes, this statement is kind of tricky. 

It's important to know that `void` binds stronger than `!==`, so this doesn't evaluate to `void (0!==n)` but to `(void 0)!==n`. `void` evaluates the following expression and always returns `undefined` afterwards, so `void 0`also evaluates to `undefined`. This `void` is probably only used here to access the `undefined` value, since it was possible to actually call a variable `undefined` in earlier JS versions, which would shadow the `undefined` value. 

So basically this statement is identical to 
`sliderMin=Number(null!==(n=this.state.numericMin)&&undefined!==n?n:this.state.sliderMin)`
which should work as desired. If `numericMin` is 0, the overall expression will still return 0. 

There is actually a semantic change in case `this.state.numericMin` is `undefined`, because in the previous code this would result in `this.state.numericMin` being returned, since we only check for `null` there. However, I guess there probably shouldn't be a case where the value is actually `undefined` any way - and even if there was, the solution with `??` would still be preferential, since we certainly don't want to have `Number(undefined)` being evaluated.

Reply:
WoW, JS is tricky :-) Thanks for the explanation. I guess `void 0` could also be used because it is shorter than `undefined`. And indeed, checking for `null` and `undefined` makes more sense.